service.rb

```diff
@@ -33,29 +35,13 @@ def initialize(port)
       when ["POST", "/add/data"]
         content_type = "text/html"
         response_status_code = "201 Created"
+        response_message = ""
 
-        while true
-          line = client.readline
-          if line.include?("Content-Length: ")
-            line.slice!("Content-Length: ")
-            content_length = line.to_i
-            break
-          end
-        end
-
-        while true
-          line = client.readline
-          if line == "\r\n"
-            body = client.read(content_length)
-            break
-          end
-        end
+        headers = HttpRequest.headers(client)
+        body = headers["Message Body"]
```

Comment:
I get what you’re going for here, but this line could’ve just been `body = client.read(headers['Content-Length'].to_i)` rather than trying to smuggle the body out in the `headers` hash.

Reply:
gosh ok true!

Reply:
ah right cause the body is not technically part of the header duh doi